### PR TITLE
Feature/iii-3590 - menubar and iframe

### PR DESCRIPTION
**Pull request by simon-debruijn**
### Added

- Icon component
- Icon story
- SideBar
- Iframe
- redirecting routes 
- part of login flow

---

Ticket: https://jira.uitdatabank.be/browse/III-3590

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,14 +21,19 @@
...
+    "@fortawesome/fontawesome-svg-core": "^1.2.32",
+    "@fortawesome/free-brands-svg-icons": "^5.15.1",
+    "@fortawesome/free-solid-svg-icons": "^5.15.1",
+    "@fortawesome/react-fontawesome": "^0.1.12",
     "bootstrap": "^4.5.3",
     "i18next": "^19.8.3",
     "i18next-browser-languagedetector": "^6.0.1",
     "lodash": "^4.17.20",
-    "next": "^10.0.0",
+    "next": "10.0.1",
```
Comment (geoffreydhuyvetters) [inline package.json:32]: Why are we removing the semver range here?

File: src/components/SideBar.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                  'hover.backgroundColor',
+                )};
+              }
+            }
+          `}
+          onClick={onClick}
+        >
+          <Content iconName={iconName}>{children}</Content>
+        </Button>
+      )}
```
Comment (geoffreydhuyvetters) [inline src/components/SideBar.js:92]: Not sure if it makes sense to change MenuItem for 1 scenario. a completely custom menu for the notifications would make more sense imo

File: src/components/SideBar.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+          > :not(:first-child) {
+            border-top: 1px solid ${getValueForMenu('borderColor')};
+          }
+        `}
+      >
+        <Menu items={userMenu} />
+        {/* TODO: this should work 'justifyContent="space-between"' */}
+        <Stack css="flex: 1; justify-content: space-between;">
+          <Menu items={manageMenu} title={t('menu.management')} />
+          <Menu items={notificationMenu} />
```
Comment (geoffreydhuyvetters) [inline src/components/SideBar.js:242]: See comment above.

File: src/components/SideBar.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        css={`
+          flex: 1;
+          > :not(:first-child) {
+            border-top: 1px solid ${getValueForMenu('borderColor')};
+          }
+        `}
+      >
+        <Menu items={userMenu} />
+        {/* TODO: this should work 'justifyContent="space-between"' */}
+        <Stack css="flex: 1; justify-content: space-between;">
```
Comment (geoffreydhuyvetters) [inline src/components/SideBar.js:240]: Should investigate this, because this should work.

File: src/components/publiq-ui/Link.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ const Link = ({ href, children, as, className, ...props }) => {
 
   if (isInternalLink) {
     return (
-      <NextLink prefetch href={href} passHref>
+      <NextLink href={href} passHref>
```
Comment (geoffreydhuyvetters) [inline src/components/publiq-ui/Link.js:25]: Why did you remove the prefetch here?

File: src/layouts/default.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      </Box>
+    </Inline>
+  );
+};
+
+Default.propTypes = {
+  children: PropTypes.node,
+};
+
+export default Default;
```
Comment (geoffreydhuyvetters) [inline src/layouts/default.js:28]: I think this makes sense to name default in Nuxt, because Nuxt automatically loads it as the default template.

But, it looks really weird when you use the component yourself. Could we give it a more descriptive name?

File: src/pages/_app.js [hunk 7]
```diff
@@ -3,14 +3,19 @@ import '../styles/global.scss';
...
-    </ThemeProvider>
-  </I18nextProvider>
-);
+const App = ({ Component, pageProps }) => {
+  return (
+    <I18nextProvider i18n={i18n}>
+      <ThemeProvider>
+        <CookiesProvider>
+          <Component {...pageProps} />
+        </CookiesProvider>
```
Comment (geoffreydhuyvetters) [inline src/pages/_app.js:15]: I guess choosing the cookie solution depends a lot of whether we need this Provider here (mainly for SSR reasons). If we want to go for a 100% SPA approach, we need to configure Next a bit more https://colinhacks.com/essays/building-a-spa-with-nextjs

File: src/pages/login/nl.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  setCookie('udb-language', 'nl', cookieOptions);
+
+  return (
+    <div>
+      <LoginButton language="nl" />
+    </div>
+  );
+};
+
+export default Nl;
```
Comment (geoffreydhuyvetters) [inline src/pages/login/nl.js:19]: nl and fr should be the same page but with different copy, can't we solve this by using a [...params].js file?

File: public/assets/udb-logo-mobile.svg [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+      />
+    </g>
+</svg>
```
Comment (geoffreydhuyvetters) [inline public/assets/udb-logo-mobile.svg:16]: We can make React components that render SVG code out of them, that way we don't have to load them via image tags, same the component below

File: next.config.js [hunk 10]
```diff
@@ -5,4 +5,22 @@ module.exports = {
...
+  },
+  env: {
+    API_KEY: process.env.API_KEY,
+    API_URL: process.env.API_URL,
+    LEGACY_APP_URL: process.env.LEGACY_APP_URL,
+    AUTH_URL: process.env.AUTH_URL,
+    SOCKET_URL: process.env.SOCKET_URL,
+    NEW_FEATURES_URL: process.env.NEW_FEATURES_URL,
+    ENV_NAME: process.env.ENV_NAME,
+  },
```
[the remark this answers is not in the document]
Comment (simon-debruijn) [reply next.config.js:25]: I wasn't able to make it work by only adding .local, found this as the solution

File: src/components/LoginButton.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
+import { useCookies } from 'react-cookie';
```
[the remark this answers is not in the document]
Comment (simon-debruijn) [reply src/components/LoginButton.js:2]: We don't I picked this one because it matches well with the hooks-philosophy of React. But I was also worried this will possibly have to be replaced by another package if we want to use the cookies in plain javascript files

File: next.config.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (geoffreydhuyvetters) [reply next.config.js:25]: should work just fine, let's investigate together

File: src/components/LoginButton.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (geoffreydhuyvetters) [reply src/components/LoginButton.js:2]: Not every function in a component needs to be a hook, what would be the advantage of using a hook here? Does it automatically handle internal state or re-renders?

File: next.config.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (geoffreydhuyvetters) [reply next.config.js:25]: https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

File: src/components/publiq-ui/Link.js (hunk 5 above)
Comment (simon-debruijn) [reply src/components/publiq-ui/Link.js:25]: I got console info that prefetch wasn't needed anymore

Comment (geoffreydhuyvetters) [reply src/components/publiq-ui/Link.js:25]: Oh, nice. Probably new since 10 then.
